packages/dashboard/src/components/sidePanel/sections/inputSettingsSection/index.spec.tsx

```diff
@@ -0,0 +1,45 @@
+import { InputWidget } from '../../../../types';
+import { MOCK_INPUT_WIDGET } from '../../../../../testing/mocks';
+import { DashboardState } from '../../../../store/state';
+import { Provider } from 'react-redux';
+import { configureDashboardStore } from '../../../../store';
+import { DefaultDashboardMessages } from '../../../../messages';
+import React from 'react';
+import { render } from '@testing-library/react';
+import InputSettings from '../inputSettingsSection';
+import wrapper from '@cloudscape-design/components/test-utils/dom';
+
+const widget: InputWidget = { ...MOCK_INPUT_WIDGET };
+
+const state: Partial<DashboardState> = {
+  dashboardConfiguration: {
+    widgets: [widget],
+    viewport: { duration: '5m' },
+  },
+  selectedWidgets: [widget],
+};
+
+const TestComponent = () => (
+  <Provider store={configureDashboardStore(state)}>
+    <InputSettings messageOverride={DefaultDashboardMessages} />
+  </Provider>
+);
+
+it('renders', () => {
```

Comment:
these tests don't give too much confidence that the component actually works. This is more or less a snapshot test with some indirection via testIds

A better test would be that as a user, i can add an option, and then see that newly added option reflected in the dashboard definition.

I would advise to avoid adding testid's on every element.

Reply:
I agree on testing component functionality, but would agree with Norbert on his use of test ids. 

I'd want to see `.findButton('[data-test-id="input-widget-add-option-btn"]').click()` instead of only truthy assertions, unless we're checking that something rendered only.

Also, defining an enum or object to hold the testing ids - that's scoped to the component - would be an easier read.